flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerImplTest.java

```diff
@@ -18,166 +18,218 @@
+	@Test
+	public void testGetOperatorBackPressureStatsAfterShutDown() throws Exception {
+		final ExecutionJobVertex jobVertex = BackPressureTrackerTestUtils.createExecutionJobVertex();
 
-		// Verify the stats
-		Assert.assertEquals(sampleId, stats.getSampleId());
-		Assert.assertEquals(endTime, stats.getEndTimestamp());
-		Assert.assertEquals(taskVertices.length, stats.getNumberOfSubTasks());
+		final int requestId = 0;
+		final long startTime = System.currentTimeMillis();
+		final long endTime = startTime + 1;
+		final double backPressureRatio = 0.1;
 
-		for (int i = 0; i < taskVertices.length; i++) {
-			double ratio = stats.getBackPressureRatio(i);
-			// Traces until sub task index are back pressured
-			Assert.assertEquals((i + 1) / ((double) 4), ratio, 0.0);
-		}
+		final BackPressureStats backPressureStats = createBackPressureStats(
+			jobVertex.getTaskVertices(), requestId, startTime, endTime, backPressureRatio);
+		final BackPressureStatsTracker tracker = createBackPressureTracker(600000, 10000, backPressureStats);
+
+		tracker.shutDown();
+
+		// trigger back pressure stats request
+		tracker.getOperatorBackPressureStats(jobVertex);
+
+		assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
 	}
 
-	private StackTraceElement[] createStackTrace(boolean isBackPressure) {
-		if (isBackPressure) {
-			return new StackTraceElement[] { new StackTraceElement(
-					BackPressureStatsTrackerImpl.EXPECTED_CLASS_NAME,
-					BackPressureStatsTrackerImpl.EXPECTED_METHOD_NAME,
-					"LocalBufferPool.java",
-					133) };
-		} else {
-			return Thread.currentThread().getStackTrace();
-		}
+	@Test
+	public void testStatsCleanup() throws Exception {
+		final ExecutionJobVertex jobVertex = BackPressureTrackerTestUtils.createExecutionJobVertex();
+		final int cleanUpInterval = 2000;
+		final long waitingTime = cleanUpInterval + 500;
+
+		final int requestId = 0;
+		final long startTime = System.currentTimeMillis();
+		final long endTime = startTime + 1;
+		final double backPressureRatio = 0.1;
+
+		final BackPressureStats backPressureStats = createBackPressureStats(
+			jobVertex.getTaskVertices(), requestId, startTime, endTime, backPressureRatio);
+		final BackPressureStatsTracker tracker = createBackPressureTracker(2000, 10000, backPressureStats);
+
+		// trigger back pressure stats request
+		tracker.getOperatorBackPressureStats(jobVertex);
+
+		Optional<OperatorBackPressureStats> optionalStats = tracker.getOperatorBackPressureStats(jobVertex);
+		assertTrue(optionalStats.isPresent());
+		OperatorBackPressureStats stats = optionalStats.get();
+
+		checkOperatorBackPressureStats(jobVertex.getTaskVertices(), requestId, endTime, backPressureRatio, stats);
+
+		tracker.cleanUpOperatorStatsCache();
+		assertTrue(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
+
+		// wait until we are ready to cleanup
+		Thread.sleep(waitingTime);
+
+		tracker.cleanUpOperatorStatsCache();
```

Comment:
Actually we do not need cleanup again.

Reply:
In this test, there is no async cleanup thread running, so we need to call the cleanup method explicitly.